drivers/sensor/hdc20x0/hdc20x0.c

```diff
@@ -0,0 +1,246 @@
+/* Driver for TI hdc20x0
+ * hdc20x0 ±2% digital humidity sensor with temperature sensor */
+
+/*
+ * Copyright (c) 2020 seeed.cc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_hdc20x0
+
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <init.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+
+#include <drivers/i2c.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(hdc20x0, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Register addresses as in the datasheet */
+#define hdc20x0_REG_TEMP_LOW 0x00
+#define hdc20x0_REG_TEMP_HIGH 0x01
+#define hdc20x0_REG_HUMIDITY_LOW 0x02
+#define hdc20x0_REG_HUMIDITY_HIGH 0x03
+#define hdc20x0_REG_INTERRUPT_DRDY 0x04
+#define hdc20x0_REG_TEMP_MAX 0x05
+#define hdc20x0_REG_HUMIDITY_MAX 0x06
+#define hdc20x0_REG_INTERRUPT_EN 0x07
+#define hdc20x0_REG_TEMP_OFFSET_ADJ 0x08
+#define hdc20x0_REG_HUMIDITY_OFFSET_ADJ 0x09
+#define hdc20x0_REG_TEMP_THR_L 0x0a
+#define hdc20x0_REG_TEMP_THR_H 0x0b
+#define hdc20x0_REG_RH_THR_L 0x0c
+#define hdc20x0_REG_RH_THR_H 0x0d
+#define hdc20x0_REG_RESET_DRDY_INT_CONF 0x0e
+#define hdc20x0_REG_MEASUREMENT_CONF 0x0f
+
+#define hdc20x0_MEAS_CONF GENMASK(2, 1)
+#define hdc20x0_MEAS_TRIG BIT(0)
+#define hdc20x0_HEATER_EN BIT(3)
+#define hdc20x0_AMM GENMASK(6, 4)
+
+struct hdc20x0_data {
+	const struct device *i2c_master;
+
+	/* Temperature Values */
+	int32_t temp_val1;
+	int32_t temp_val2;
+	/* Humidity Values*/
+	int32_t humidity_val1;
+	int32_t humidity_val2;
+};
+
+struct hdc20x0_cfg {
+	char *i2c_bus_name;
+	uint16_t i2c_addr;
+};
+
+static int
+hdc20x0_read_measurement_temperature(struct hdc20x0_data *data,
```

Comment:
Shorter identifiers would be appropriate, for example hdc20xx_read_temp(...), which is still understandable enough.